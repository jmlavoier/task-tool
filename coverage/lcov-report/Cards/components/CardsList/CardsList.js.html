<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Cards/components/CardsList/CardsList.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">Cards/components/CardsList</a> CardsList.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/5</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import uid from 'uid';
&nbsp;
import Card from 'modules/Cards/components/Card';
import BtnAddCard from 'modules/Cards/components/BtnAddCard';
&nbsp;
import style from './CardsList.sass';
&nbsp;
const CardsList = <span class="fstat-no" title="function not covered" >({</span> cards, boardId, onBtnAddCardClick, onSaveCard, onEditCard, changeBoard }) =&gt; (
<span class="cstat-no" title="statement not covered" >  &lt;ul className={style['cards-list']}&gt;</span>
    {cards.map(<span class="fstat-no" title="function not covered" >({</span> id, description, editMode }) =&gt; (<span class="cstat-no" title="statement not covered" >&lt;Card</span>
      key={uid()}
      id={id}
      description={description}
      onSaveCard={onSaveCard}
      onEditCard={onEditCard}
      editMode={editMode}
      changeBoard={changeBoard}
    /&gt;))}
    &lt;BtnAddCard onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >onBtnAddCardClick(boardId)}</span> /&gt;
  &lt;/ul&gt;
);
&nbsp;
CardsList.propTypes = {
  cards: PropTypes.arrayOf(PropTypes.object).isRequired,
  boardId: PropTypes.string.isRequired,
  onBtnAddCardClick: PropTypes.func.isRequired,
  onSaveCard: PropTypes.func.isRequired,
  onEditCard: PropTypes.func.isRequired,
  changeBoard: PropTypes.func.isRequired,
};
&nbsp;
export default CardsList;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Nov 04 2017 16:51:37 GMT-0200 (-02)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
