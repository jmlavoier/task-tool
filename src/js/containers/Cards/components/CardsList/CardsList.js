import React from 'react';
import PropTypes from 'prop-types';

import Card from 'containers/Cards/components/Card';
import BtnAddCard from 'containers/Cards/components/BtnAddCard';

import style from './CardsList.sass';

const CardsList = ({ cards, boardId, onBtnAddCardClick, onSaveCard, onEditCard, changeBoard }) => {
  return (
  <ul className={style['cards-list']}>
    {cards.map(({ id, description, editMode }, i) => <Card 
      key={i} 
      id={id}
      description={description} 
      onSaveCard={onSaveCard} 
      onEditCard={onEditCard}
      editMode={editMode}
      changeBoard={changeBoard}
    />)}
    <BtnAddCard onClick={() => onBtnAddCardClick(boardId)} />
  </ul>
)};

CardsList.propTypes = {
  cards: PropTypes.array.isRequired,
  onBtnAddCardClick: PropTypes.func.isRequired, 
}

export default CardsList;