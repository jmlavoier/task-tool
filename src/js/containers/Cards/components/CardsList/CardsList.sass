.cards-list
  padding-left: 0