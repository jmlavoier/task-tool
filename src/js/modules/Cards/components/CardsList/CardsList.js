import React from 'react';
import PropTypes from 'prop-types';

import Card from 'modules/Cards/components/Card';
import BtnAddCard from 'modules/Cards/components/BtnAddCard';

import style from './CardsList.sass';

const CardsList = ({ cards, boardId, onBtnAddCardClick, onSaveCard, onEditCard, changeBoard }) => (
  <ul className={style['cards-list']}>
    {cards.map(({ id, description, editMode }, i) => (<Card
      key={i}
      id={id}
      description={description}
      onSaveCard={onSaveCard}
      onEditCard={onEditCard}
      editMode={editMode}
      changeBoard={changeBoard}
    />))}
    <BtnAddCard onClick={() => onBtnAddCardClick(boardId)} />
  </ul>
);

CardsList.propTypes = {
  cards: PropTypes.array.isRequired,
  onBtnAddCardClick: PropTypes.func.isRequired,
};

export default CardsList;
