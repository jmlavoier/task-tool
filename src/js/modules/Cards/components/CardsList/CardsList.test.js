import React from 'react';
import { shallow, mount } from 'enzyme';
import TestBackend from 'react-dnd-test-backend';
import { DragDropContext } from 'react-dnd';

import CardsList from './CardsList';

function wrapInTestContext(DecoratedComponent) {
  return DragDropContext(TestBackend)(props => <DecoratedComponent {...props} />);
}

const OriginalBox = wrapInTestContext(CardsList);
const identity = el => el;

const mockCards = [
  { id: '1', description: '1', editMode: false },
  { id: '2', description: '2', editMode: false },
  { id: '3', description: '3', editMode: false },
];
const boardId = '321';

test('Should component render with default props', () => {
  const wrapper = mount(<OriginalBox
    cards={[]}
    boardId={''}
    onBtnAddCardClick={() => {}}
    onSaveCard={() => {}}
    onEditCard={() => {}}
    changeBoard={() => {}}
  />);

  expect(wrapper).toMatchSnapshot();
});

test('Should component render cards', () => {
  const wrapper = mount(<OriginalBox
    cards={mockCards}
    boardId={boardId}
    onBtnAddCardClick={() => {}}
    onSaveCard={() => {}}
    onEditCard={() => {}}
    changeBoard={() => {}}
  />);

  expect(wrapper).toMatchSnapshot();
});

test('Should call onBtnAddCardClick', () => {
  const mockBtnAddCardClick = jest.fn();

  const wrapper = mount(<OriginalBox
    cards={mockCards}
    boardId={boardId}
    onBtnAddCardClick={mockBtnAddCardClick}
    onSaveCard={() => {}}
    onEditCard={() => {}}
    changeBoard={() => {}}
  />);

  wrapper.find('.btn-add-card').simulate('click');

  expect(mockBtnAddCardClick).toHaveBeenCalled();
  expect(mockBtnAddCardClick.mock.calls[0][0]).toEqual(boardId);
});