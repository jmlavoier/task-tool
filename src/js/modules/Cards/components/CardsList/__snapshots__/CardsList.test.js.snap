// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should component render cards 1`] = `
<DragDropContext(Component)
  boardId="321"
  cards={
    Array [
      Object {
        "description": "1",
        "editMode": false,
        "id": "1",
      },
      Object {
        "description": "2",
        "editMode": false,
        "id": "2",
      },
      Object {
        "description": "3",
        "editMode": false,
        "id": "3",
      },
    ]
  }
  changeBoard={[Function]}
  onBtnAddCardClick={[Function]}
  onEditCard={[Function]}
  onSaveCard={[Function]}
>
  <Component
    boardId="321"
    cards={
      Array [
        Object {
          "description": "1",
          "editMode": false,
          "id": "1",
        },
        Object {
          "description": "2",
          "editMode": false,
          "id": "2",
        },
        Object {
          "description": "3",
          "editMode": false,
          "id": "3",
        },
      ]
    }
    changeBoard={[Function]}
    onBtnAddCardClick={[Function]}
    onEditCard={[Function]}
    onSaveCard={[Function]}
  >
    <CardsList
      boardId="321"
      cards={
        Array [
          Object {
            "description": "1",
            "editMode": false,
            "id": "1",
          },
          Object {
            "description": "2",
            "editMode": false,
            "id": "2",
          },
          Object {
            "description": "3",
            "editMode": false,
            "id": "3",
          },
        ]
      }
      changeBoard={[Function]}
      onBtnAddCardClick={[Function]}
      onEditCard={[Function]}
      onSaveCard={[Function]}
    >
      <ul
        className="cards-list"
      >
        <DragSource(Card)
          changeBoard={[Function]}
          description="1"
          editMode={false}
          id="1"
          onEditCard={[Function]}
          onSaveCard={[Function]}
        >
          <Card
            changeBoard={[Function]}
            connectDragSource={[Function]}
            description="1"
            editMode={false}
            id="1"
            isDragging={false}
            onEditCard={[Function]}
            onSaveCard={[Function]}
          >
            <div
              className="card"
              onClick={[Function]}
              role="presentation"
            >
              1
            </div>
          </Card>
        </DragSource(Card)>
        <DragSource(Card)
          changeBoard={[Function]}
          description="2"
          editMode={false}
          id="2"
          onEditCard={[Function]}
          onSaveCard={[Function]}
        >
          <Card
            changeBoard={[Function]}
            connectDragSource={[Function]}
            description="2"
            editMode={false}
            id="2"
            isDragging={false}
            onEditCard={[Function]}
            onSaveCard={[Function]}
          >
            <div
              className="card"
              onClick={[Function]}
              role="presentation"
            >
              2
            </div>
          </Card>
        </DragSource(Card)>
        <DragSource(Card)
          changeBoard={[Function]}
          description="3"
          editMode={false}
          id="3"
          onEditCard={[Function]}
          onSaveCard={[Function]}
        >
          <Card
            changeBoard={[Function]}
            connectDragSource={[Function]}
            description="3"
            editMode={false}
            id="3"
            isDragging={false}
            onEditCard={[Function]}
            onSaveCard={[Function]}
          >
            <div
              className="card"
              onClick={[Function]}
              role="presentation"
            >
              3
            </div>
          </Card>
        </DragSource(Card)>
        <BtnAddCard
          onClick={[Function]}
        >
          <button
            className="btn-add-card"
            onClick={[Function]}
          >
             + Add Card 
          </button>
        </BtnAddCard>
      </ul>
    </CardsList>
  </Component>
</DragDropContext(Component)>
`;

exports[`Should component render with default props 1`] = `
<DragDropContext(Component)
  boardId=""
  cards={Array []}
  changeBoard={[Function]}
  onBtnAddCardClick={[Function]}
  onEditCard={[Function]}
  onSaveCard={[Function]}
>
  <Component
    boardId=""
    cards={Array []}
    changeBoard={[Function]}
    onBtnAddCardClick={[Function]}
    onEditCard={[Function]}
    onSaveCard={[Function]}
  >
    <CardsList
      boardId=""
      cards={Array []}
      changeBoard={[Function]}
      onBtnAddCardClick={[Function]}
      onEditCard={[Function]}
      onSaveCard={[Function]}
    >
      <ul
        className="cards-list"
      >
        <BtnAddCard
          onClick={[Function]}
        >
          <button
            className="btn-add-card"
            onClick={[Function]}
          >
             + Add Card 
          </button>
        </BtnAddCard>
      </ul>
    </CardsList>
  </Component>
</DragDropContext(Component)>
`;
